libs/zzz/formula/src/data/wengine/sheets/BashfulDemon.ts

```diff
@@ -0,0 +1,54 @@
+import { prod, subscript } from '@genshin-optimizer/pando/engine'
+import type { WengineKey } from '@genshin-optimizer/zzz/consts'
+import {
+  allNumConditionals,
+  own,
+  ownBuff,
+  registerBuff,
+  teamBuff,
+} from '../../util'
+import {
+  cmpSpecialtyCount,
+  entriesForWengine,
+  registerWengine,
+  showSpecialtyCount,
+} from '../util'
+
+const key: WengineKey = 'BashfulDemon'
+const { modification } = own.wengine
+
+const { launch_ex_attack } = allNumConditionals(key, true, 0, 4)
+
+const sheet = registerWengine(
+  key,
+  // Handles base stats and passive buffs
+  entriesForWengine(key),
+
+  // passive buffs
+  registerBuff(
+    'ice_dmg_',
+    ownBuff.combat.dmg_.ice.add(
+      cmpSpecialtyCount(
+        key,
+        subscript(modification, [-1, 0.15, 0.175, 0.2, 0.22, 0.24])
```

Comment:
Can maybe think of building a pipeline that extracts numbers from string? Idk

Reply:
could be considered in the future